tower-http/src/services/fs/serve_dir.rs

```diff
@@ -228,11 +267,37 @@ fn append_slash_on_path(uri: Uri) -> Uri {
 }
 
 enum Output {
-    File(File, HeaderValue, usize),
+    File(FileRequest),
     Redirect(HeaderValue),
     NotFound,
 }
 
+struct FileRequest {
+    file: File,
+    total_size: u64,
+    chunk_size: usize,
+    mime_header_value: HeaderValue,
+    range: Option<ParsedRangeHeader>,
+}
+
+
+enum ParsedRangeHeader {
+    Valid(ContentRange),
+    Unsatisfiable,
+    Malformed(String)
+}
+
+struct ContentRange {
+    start_inclusive: u64,
+    end_inclusive: u64,
+}
+
+impl ContentRange {
+    fn range_size(&self) -> u64 {
+        self.end_inclusive - self.start_inclusive + 1
+    }
+}
```

Comment:
Could we use https://doc.rust-lang.org/stable/std/ops/struct.RangeInclusive.html instead of this?